src/services/playlist.ts

```diff
@@ -4,15 +4,265 @@
  */
 
 import { Logging } from "../logger";
-import { Autowired } from "../service";
+import { Autowired, Service } from "../service";
+import { DiscordService } from "./discord";
 import { DatabaseService } from "./database";
+import { joinVoiceChannel, VoiceConnectionStatus, createAudioPlayer, AudioPlayer, NoSubscriberBehavior, createAudioResource, AudioPlayerStatus, AudioResource, getVoiceConnection, VoiceConnection, PlayerSubscription, entersState } from '@discordjs/voice';
+import { Collection, VoiceChannel, VoiceState } from "discord.js";
 
+import config from '../config.json';
+import { Util } from "../util";
+import fs from 'fs';
+import type { Song } from "@prisma/client";
+
+@Service ()
 export class PlaylistService extends Logging {
```

Comment:
was this class tested for scenarios:

- bot is kicked from a channel by a moderator
- bot is moved to another channel by a moderator